tawazi/_dag/dag.py

```diff
@@ -417,25 +416,112 @@ def _assign_compound_priority(self) -> None:
+        fig_name: str,
+        show_nodes_names: bool = False,
+        layout_algorithm: str = "tree",
+        display: bool = True,
+    ) -> None:
         """Draws the Networkx directed graph.
 
+        Note: install tawazi with extras [draw].
+
         Args:
-            k (float): parameter for the layout of the graph, the higher, the further the nodes apart. Defaults to 0.8.
-            display (bool): display the layout created. Defaults to True.
-            t (int): time to display in seconds. Defaults to 3.
+            fig_name (str): save in local file path if valid input
+            show_nodes_names (bool): whether to display nodes names over datapoints
+            layout_algorithm (str): layout algorithm for igraph, defaults to kk.
+            display (bool): display figure
         """
-        import matplotlib.pyplot as plt
+        import igraph as ig
+        from plotly.graph_objs import Figure, Layout, Scatter, layout
+
+        ig_graph = ig.Graph(directed=True)
+        ig_graph.add_vertices(list(self.graph_ids.nodes()))
+        ig_graph.add_edges(list(self.graph_ids.edges()))
+        labels = list(ig_graph.vs["name"])
+        n_labels = len(labels)
+        edges = [e.tuple for e in ig_graph.es]  # list of edges
+        ig_layout = ig_graph.layout(layout_algorithm)  # kamada_kawai layout
+        # check https://python.igraph.org/en/stable/tutorial.html#layout-algorithms
+
+        x_nodes = [ig_layout[k][0] for k in range(n_labels)]
+        y_nodes = [ig_layout[k][1] for k in range(n_labels)]
+        x_edges = []
+        y_edges = []
+        for e in edges:
+            x_edges += [ig_layout[e[0]][0], ig_layout[e[1]][0], None]
+            y_edges += [ig_layout[e[0]][1], ig_layout[e[1]][1], None]
+
+        trace1 = Scatter(
+            x=x_edges,
+            y=y_edges,
+            name="edges",
+            mode="lines+markers",
+            line=dict(color="rgb(210,210,210)", width=2),
+            hoverinfo="none",
+            marker=dict(symbol="arrow-bar-up", size=10, angleref="previous", color="rgb(0,0,0)"),
+        )
+        nodes_display_mode = "markers"
+        if show_nodes_names:
+            nodes_display_mode = "markers+text"
+        trace2 = Scatter(
+            x=x_nodes,
+            y=y_nodes,
+            mode=nodes_display_mode,
+            name="nodes",
+            marker=dict(
+                symbol="circle-dot",
+                size=10,
+                color="#6959CD",
+                line=dict(color="rgb(50,50,50)", width=1),
+            ),
+            text=labels,
+            textposition="top center",
+            hoverinfo="text",
+        )
 
-        # TODO: use graphviz instead! it is much more elegant
+        axis = dict(
+            showline=False,  # hide axis line, grid, ticklabels and  title
+            zeroline=False,
+            showgrid=False,
+            showticklabels=False,
+            title="",
+        )
+
+        width = 1200
```

Comment:
these should be optionally parametrizable

Reply:
nice catch